core/equations.gms

```diff
@@ -838,16 +832,65 @@ q_limitCO2(ttot+1,regi) $((pm_ttot_val(ttot+1) ge max(cm_startyear,2055)) AND (t
          =l=
          vm_emiTe(ttot,regi,"co2");
 
+***---------------------------------------------------------------------------
+*' Adjustment costs - calculation of the relative change to last time step 
+***---------------------------------------------------------------------------
+
 q_eqadj(regi,ttot,teAdj(te))$(ttot.val ge max(2010, cm_startyear)) ..
-         v_adjFactor(ttot,regi,te)
-         =e=
-         power(
-         (sum(te2rlf(te,rlf),vm_deltaCap(ttot,regi,te,rlf)) - sum(te2rlf(te,rlf),vm_deltaCap(ttot-1,regi,te,rlf)))/(pm_ttot_val(ttot)-pm_ttot_val(ttot-1))
-         ,2)
-                /( sum(te2rlf(te,rlf),vm_deltaCap(ttot-1,regi,te,rlf)) + p_adj_seed_reg(ttot,regi) * p_adj_seed_te(ttot,regi,te)  
-                   + p_adj_deltacapoffset("2010",regi,te)$(ttot.val eq 2010) + p_adj_deltacapoffset("2015",regi,te)$(ttot.val eq 2015)
-                   + p_adj_deltacapoffset("2020",regi,te)$(ttot.val eq 2020) + p_adj_deltacapoffset("2025",regi,te)$(ttot.val eq 2025)
-                  );
+  v_adjFactor(ttot,regi,te)
+  =e=
+  power(
+    ( sum(te2rlf(te,rlf), vm_deltaCap(ttot,regi,te,rlf)) - sum(te2rlf(te,rlf), vm_deltaCap(ttot-1,regi,te,rlf)) )
+    / ( pm_ttot_val(ttot) - pm_ttot_val(ttot-1) )
+  , 2)
+  / ( sum(te2rlf(te,rlf), vm_deltaCap(ttot-1,regi,te,rlf)) + p_adj_seed_reg(ttot,regi) * p_adj_seed_te(ttot,regi,te)  
+      + p_adj_deltacapoffset("2010",regi,te)$(ttot.val eq 2010) + p_adj_deltacapoffset("2015",regi,te)$(ttot.val eq 2015)
+      + p_adj_deltacapoffset("2020",regi,te)$(ttot.val eq 2020) + p_adj_deltacapoffset("2025",regi,te)$(ttot.val eq 2025)
+    )
+;
+
+***---------------------------------------------------------------------------
+*' Calculate changes to REF in cm_startyear - needed to limit them via refunded adj costs 
+***---------------------------------------------------------------------------
+*' calculating the absolute change of output with respect to the value in REF for each te (counting SE, FE, UE and CCS)
+q_changeProdStartyear(t,regi,te)$( (t.val gt 2005) AND (t.val eq cm_startyear ) )..
+  v_changeProdStartyear(t,regi,te)
+  =e=
+  sum(pe2se(enty,enty2,te),   vm_prodSe(t,regi,enty,enty2,te)  - p_prodSeReference(t,regi,enty,enty2,te) )
+  + sum(se2se(enty,enty2,te), vm_prodSe(t,regi,enty,enty2,te)  - p_prodSeReference(t,regi,enty,enty2,te) )
+  + sum(se2fe(enty,enty2,te), vm_prodFE(t,regi,enty,enty2,te)  - p_prodFEReference(t,regi,enty,enty2,te) )
+  + sum(fe2ue(enty,enty2,te), vm_prodUe(t,regi,enty,enty2,te)  - p_prodUeReference(t,regi,enty,enty2,te) )
+  + sum(ccs2te(enty,enty2,te), sum(teCCS2rlf(te,rlf), vm_co2CCS(t,regi,enty,enty2,te,rlf) - p_co2CCSReference(t,regi,enty,enty2,te,rlf) ) )
+;
+
+*' calculating the relative change 
+q_relChangeProdStartYear(t,regi,te)$( (t.val gt 2005) AND (t.val eq cm_startyear ) )..
```

Comment:
could you move the denominator of the division to the left hand side of the equation as a multiplication term?
I am not sure this will ever be the case, but just to be on the safe side we avoid divisions by zero error in extreme cases, like for example very small numbers in the denominator. So it is easier to just transform the equation into multiplication on both sides instead of adding a division term, as `10^-12 * 0` is defined, meanwhile `# / 10^-12` can be not defined depending of the zero tolerance we set in gams.

Reply:
in this case the parameters "p_adj_seed_reg(t,regi) * p_adj_seed_te(t,regi,te)" should ensure that the denominator never gets very small, but I agree that it is always better to formulate relative variables with a multiplication, so I changed as you proposed.